packages/repository-json-schema/src/build-schema.ts

```diff
@@ -91,46 +96,90 @@ export function modelToJsonSchema(ctor: Function): JsonDefinition {
         }
       }
     }
-    if (propCtor && typeof propCtor === 'function') {
-      // errors out if @property.array() is not used on a property of array
-      if (propCtor === Array) {
-        throw new Error('type is defined as an array');
-      }
 
-      const propDef: JsonDefinition = determinePropertyDef(propCtor);
+    // errors out if @property.array() is not used on a property of array
+    if (propCtor === Array) {
+      throw new Error('type is defined as an array');
+    }
 
-      if (!schema.properties) {
-        schema.properties = {};
-      }
+    const propDef: JsonDefinition = determinePropertyDef(propCtor);
 
-      if (propMeta.array === true) {
-        schema.properties[p] = {
-          type: 'array',
-          items: propDef,
-        };
-      } else {
-        schema.properties[p] = propDef;
+    if (!schema.properties) {
+      schema.properties = {};
+    }
+
+    if (propMeta.validationKey) {
+      if (propMeta.validationKey === 'oneOf') {
+        let property: JsonDefinition = (schema.properties[prop] =
+          schema.properties[prop] || {});
+        if (!property.oneOf) {
+          property.oneOf = [];
+        }
+        if (propMeta.array) {
+          property.oneOf.push({type: 'array', items: propDef});
+        } else {
+          property.oneOf.push(propDef);
+        }
       }
+    } else if (propMeta.array) {
+      schema.properties[prop] = {
+        type: 'array',
+        items: propDef,
+      };
+    } else {
+      schema.properties[prop] = propDef;
+    }
 
-      if (isComplexType(propCtor)) {
-        const propSchema = getJsonSchema(propCtor);
+    // populating JSON Schema 'definitions'
+    if (isComplexType(propCtor)) {
+      const propSchema = getJsonSchema(propCtor);
 
-        if (propSchema && Object.keys(propSchema).length > 0) {
-          if (!schema.definitions) {
-            schema.definitions = {};
-          }
+      if (propSchema && Object.keys(propSchema).length > 0) {
+        if (!schema.definitions) {
+          schema.definitions = {};
+        }
 
-          if (propSchema.definitions) {
-            for (const key in propSchema.definitions) {
-              schema.definitions[key] = propSchema.definitions[key];
-            }
-            delete propSchema.definitions;
+        if (propSchema.definitions) {
+          for (const key in propSchema.definitions) {
+            schema.definitions[key] = propSchema.definitions[key];
           }
-
-          schema.definitions[propCtor.name] = propSchema;
+          delete propSchema.definitions;
         }
+
+        schema.definitions[propCtor.name] = propSchema;
+      }
+    }
+  };
+
+  if (meta.title) {
```

Comment:
The `title` property does not make sense to me. IMO, the `title` should be always set to the model name.

See e.g. http://json-schema.org/example1.html#starting-the-schema

```json
{
    "$schema": "http://json-schema.org/draft-06/schema#",
    "title": "Product",
    "description": "A product from Acme's catalog",
    "type": "object"
}
```

I am imagining such schema would be created from the following code:

```ts
@model({
  description: 'A product from Acme\'s catalog',
})
class Product {
}
```

Reply:
I don't see the harm in allowing it to be overridden, though I'd expect that to take the form of `result.title = meta.title || ctor.prototype.name;` or something similar

Reply:
Fair enough. Even though it's not required for MVP, it looks like a small addition with big benefits.

@shimks Please make sure there is a unit-test covering all different combinations (title not provided - model name used, title provided - model name is ignored) and also the documentation mentions this behavior.